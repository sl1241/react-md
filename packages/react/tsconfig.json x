{
  "extends": "../../tsconfig.base.json",
  "compilerOptions": {
    "outDir": "./dist",
    "rootDir": "./src",
    "declarationDir": "./dist/types",
    "composite": true,
    "skipLibCheck": true,
    "skipDefaultLibCheck": true,
    "jsx": "preserve",
    "jsxImportSource": "voby",
    "types": [
      "voby/dist/jsx/jsx"
    ],
  },
  "exclude": [
    "vite.config.ts",
"vite.config.umd.ts",
    "types",
  ],
}
