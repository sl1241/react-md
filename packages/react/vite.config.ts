import { defineConfig } from 'vite'

const config = defineConfig({
    build: {
        minify: false,
        lib: {
            entry: ["./src/index.ts"],
            name: "react",
            formats: ['cjs', 'es', 'umd'],
            fileName: (format: string, entryName: string) => `${entryName}.${format}.js`
        },
        // rollupOptions: {
        //   output: {
        //     preserveModules: true,
        //   }
        // },
    },


    esbuild: {
        jsx: 'automatic',
    },

    css: {
        modules: {
            scopeBehaviour: 'global',
            localsConvention: 'camelCase',
        }
    },
})



export default config
