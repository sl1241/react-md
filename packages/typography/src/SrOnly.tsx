
import { bem } from "@react-md/utils"

import type { TypographyHTMLElement, TypographyProps } from "./Typography"
import { Typography } from "./Typography"

export interface SrOnlyProps extends TypographyProps {
    /**
     * Boolean if the text should become visible when focused. If this prop is
     * enabled and the `tabIndex` prop is `undefined`, the `tabIndex` will be
     * updated to be `0`.
     */
    focusable?: FunctionMaybe<Nullable<boolean>>
}

const block = bem("rmd-sr-only")

/**
 * This component is used to create text that is only visible to screen readers.
 * If you enable the `focusable` prop, the text will become visible to all users
 * while focused.
 */
export const SrOnly = //TypographyHTMLElement(
    (
        {
            className,
            children,
            focusable = false,
            tabIndex: propTabIndex,
            component = "span",
            ref,
            ...props
        }: SrOnlyProps
    ) => {
        let tabIndex = propTabIndex
        if (focusable && typeof tabIndex === "undefined") {
            tabIndex = 0
        }

        return (
            <Typography
                {...props}
                ref={ref}
                tabIndex={tabIndex}
                component={component}
                className={[block({ focusable }), className]}
            >
                {children}
            </Typography>
        )
    }
