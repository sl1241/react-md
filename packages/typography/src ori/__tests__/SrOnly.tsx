import { render } from "@testing-library/react";

import { SrOnly } from "../SrOnly";

describe("SrOnly", () => {
  it("should render correctly", () => {
    const { container, rerender } = render(<SrOnly>Some Text</SrOnly>);
    expect(container).toMatchSnapshot();

    rerender(<SrOnly focusable>Some Text</SrOnly>);
    expect(container).toMatchSnapshot();

    rerender(<SrOnly tabIndex={-1}>Some Text</SrOnly>);
    expect(container).toMatchSnapshot();

    rerender(
      <SrOnly tabIndex={-1} focusable>
        Some Text
      </SrOnly>
    );
    expect(container).toMatchSnapshot();

    rerender(<SrOnly component="div">Some Text</SrOnly>);
    expect(container).toMatchSnapshot();
  });
});
