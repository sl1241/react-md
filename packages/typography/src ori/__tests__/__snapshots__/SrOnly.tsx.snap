// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SrOnly should render correctly 1`] = `
<div>
  <span
    class="rmd-typography rmd-typography--body-1 rmd-sr-only"
  >
    Some Text
  </span>
</div>
`;

exports[`SrOnly should render correctly 2`] = `
<div>
  <span
    class="rmd-typography rmd-typography--body-1 rmd-sr-only rmd-sr-only--focusable"
    tabindex="0"
  >
    Some Text
  </span>
</div>
`;

exports[`SrOnly should render correctly 3`] = `
<div>
  <span
    class="rmd-typography rmd-typography--body-1 rmd-sr-only"
    tabindex="-1"
  >
    Some Text
  </span>
</div>
`;

exports[`SrOnly should render correctly 4`] = `
<div>
  <span
    class="rmd-typography rmd-typography--body-1 rmd-sr-only rmd-sr-only--focusable"
    tabindex="-1"
  >
    Some Text
  </span>
</div>
`;

exports[`SrOnly should render correctly 5`] = `
<div>
  <div
    class="rmd-typography rmd-typography--body-1 rmd-sr-only"
  >
    Some Text
  </div>
</div>
`;
